crates/ruff_python_ast/src/source_code/locator.rs

```diff
@@ -2,11 +2,13 @@
 
 use once_cell::unsync::OnceCell;
 use rustpython_parser::ast::Location;
+use std::rc::Rc;
 
 use crate::types::Range;
 
 pub struct Locator<'a> {
     contents: &'a str,
+    contents_rc: Rc<str>,
```

Comment:
Adding an `Rc` means every `Locator` now performs a heap allocation. Is this used for getting "cheap" string formatting? Could we store a `Rc` source code on the formatting context instead (only paying once for the heap allocation)?

Reply:
Let me try putting this on the formatting context instead... We already only create one `Locator` per file, so in practice I believe it'd be "the same", but it'd be nice to keep this separate from the `Locator` API itself.